app/models/link_set.rb

```diff
@@ -1,42 +1,32 @@
 class LinkSet < ActiveRecord::Base
-  include Replaceable
-  include DefaultAttributes
-  include SymbolizeJSON
-
-  validate :links_are_valid
+  has_many :links
 
-  def links=(links_hash)
-    super(links_hash || {})
-  end
+  include Replaceable
 
-private
-  def self.query_keys
-    [:content_id]
-  end
+  def hashed_links
+    # This method groups LinkSet links data in a hash.
+    # Hash keys are link types and its values are arrays of target content ids.
+    # ie:
+    # {
+    #   related: [ UUID, UUID ],
+    #   organisations: [ UUID ],
+    # }
 
-  def links_are_valid
-    # Test that the `links` attribute, if set, is a hash from strings to lists
-    # of UUIDs
-    return if links.empty?
+    links_hash = Hash.new
 
-    bad_keys = links.keys.reject { |key| link_key_is_valid?(key) }
-    if bad_keys.any?
-      errors[:links] = "Invalid link types: #{bad_keys.to_sentence}"
+    links.each do |l|
+      if links_hash.has_key?(l.link_type)
+        links_hash[l.link_type] << l.target_content_id
+      else
+        links_hash[l.link_type] = [l.target_content_id]
+      end
     end
 
-    bad_values = links.values.reject { |value|
-      value.is_a?(Array) && value.all? { |content_id|
-        UuidValidator.valid?(content_id)
-      }
-    }
-    if bad_values.any?
-      errors[:links] = "must map to lists of UUIDs"
-    end
+    links_hash.symbolize_keys
   end
 
-  def link_key_is_valid?(link_key)
-    link_key.is_a?(Symbol) &&
-      link_key.to_s.match(/\A[a-z0-9_]+\z/) &&
-      link_key != :available_translations
+private
```

Comment:
Fun Ruby fact: the `private` call here doesn't work like you think it would.

See http://domon.cc/2013/12/25/private-class-methods-in-ruby/ or https://github.com/bbatsov/rubocop/issues/1367#issuecomment-58933163 for details.